### ISSUE TITLE
Support (minor) updates for GBE 2023

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/gbe-divio-djangocms-python2.7#287]
This is to track the fairly minor work for getting act & class bids looking better and easier to use.

Anything big will get a separate ticket and I'll reference that here.

Initial notes from Luna: 

- We could make the language a bit more user friendly. For example, for a new user (hasn't signed up on the site before), we ask them to create an account, then "Tell us about your stage persona". While this language logically makes sense in that performers can have more than one persona, in my experience it feels more straightforward to phrase the headline more like "Tell us about yourself". You could then add some more detail about this should describe your stage persona, and how you can create a troupe profile, in the subheading. Happy to provide more details / screen designs, just giving an overview.
- Same notes for the term "Propose an Act"...we use the word "Apply" in the nav of the site, and I think it could be confusing to not see "Apply" or "Application" also on the actual application form. We can always include more description in the subheading. Always good to ensure people know they are on the right page and keep terms consistent.
- For each form field on the application form, I think we should show the additional context notes / instructions near the form field, not in a tooltip. Just makes them easier to access and reference as the user is filling out the form.
- For signing up on the Expo site, is it absolutely necessary that we ask for all the "Update your profile" information up front? Can we ask for these things when someone has been accepted to the Expo? It's a lot of info up front and could be contributing to drop-off in the flow. I also noticed that we ask for a user's email address on both the "Create and Account" page and the "Update your Profile" page and it didn't seem to remember that I'd already supplied my email when I tested it.

**Comment by bethlakshmi** [comment]
> For each form field on the application form, I think we should show the additional context notes / instructions near the form field, not in a tooltip. Just makes them easier to access and reference as the user is filling out the form.

This is really easy - I always use the same icon as the "?" so I can just search across that and fix it. 

Questions, though, from fooling around with it...

I think we want:
- label for the field in bold (as example shows) (end of line)
- text same size, normal (not bold, not italic) (end of line)
- field below

which makes the form longer vertically on desktops, and there's no change on mobile (the format already works this way there).

Is that all good?

Also - some of these are very texty, and it seems like the text bleeds into the form, should I do anything with alignment here?  Both padding between form and text and also between form field and left side of page?

**Comment by bethlakshmi** [comment]
Other areas of hover text (low priority), that I'm wondering what to do with:

- reviewing bids, for example - this [act](https://burlesque-expo-stage.us.aldryn.io/act/review/1393).  We have two uses:
      -  we have hover text for the read-only version of the bid.  This showing the same clarifications we gave to the bidder.  It seems rough to inflict all this extra text on reviewers, who presumably know what those clarifications are most of the time?  Keep it as is, or try to find an alternative layout (could use help on this).
      - at the bottom of the page (this is specific to act reviews) - we have 2 tables of radio buttons.  The "?" hover text here clarifies what we want from judges.  I could see expanding this, and putting the help text as an additional column in the tables.  But it'll blow the tables up to the point of needing to point them 1 under the other (the middle section - the detailed reviews - is only visible to Scratch, the act coordinator), and could be a problem on mobile.  .... do we care?

**Comment by burlexpo** [comment]
**Context/ToolTips**
My suggestion is that we group Question, followed by entry field, followed by Help Text, followed by some space before the next question.  Here's an image of what I mean.

<img width="285" alt="Screen Shot 2022-09-20 at 8 40 02 PM" src="https://user-images.githubusercontent.com/5369404/191389338-77c3d14f-1ac6-4cf2-9236-782d0fb108b2.png">

**Comment by burlexpo** [comment]
**Required Levels of Information/Labels**
Can we do something like allow people to register for the site, and just require a "Display/Stage Name" and Email Address?

I think we need to make it clear the you can have a different "Persona" (?maybe this is a bad choice of word?) for each thing you can do (I.e., a persona for teaching and a different persona for performing).  Maybe we want to use "Bio"? Except it's more than a Bio, since it can have a different name and image, as well as bio.  Maybe we just need some explanatory text as to _why_ someone would want to do this?

**Comment by burlexpo** [comment]
_Same notes for the term "Propose an Act"...we use the word "Apply" in the nav of the site, and I think it could be confusing to not see "Apply" or "Application" also on the actual application form. We can always include more description in the subheading. Always good to ensure people know they are on the right page and keep terms consistent._

Yes, we should definitely use the form "Apply to Perform/Performer Application" for **Performing** and **Teaching**.   What about Volunteering? In some cases people are actually applying but in a lot of places they are signing up.

**Comment by bethlakshmi** [comment]
I just pushed an update for @burlexpo's idea of doing

<label text - with *, bold, normal labels)
<field>
<help text>

And all previously mentioned forms will show it...

It'll take some finessing w. white space, I think.  With lists of stuff, there's padding at the bottom of the list, which makes it really ambiguous what form field the help text goes to (see picture)
![Screen Shot 2022-09-21 at 9 33 07 AM](https://user-images.githubusercontent.com/4542707/191518705-5f0a9a07-8129-4932-8799-168a90c34029.png)

The arrows are how the fields & help text are actually connected...  but if you look at the test version ... is it something you'd intuit?

**Comment by bethlakshmi** [comment]
**Text Change Stuff**
FYI - I also am pushing an update to make as much text editable as possible in these areas.

![Screen Shot 2022-09-21 at 9 49 57 AM](https://user-images.githubusercontent.com/4542707/191556964-5624af8e-cba4-4a58-9a54-aa4cc450f7b7.png)


The 4 areas with named circles pointing to them are editable in "/admin/gbe/usermessage/" on any given site (reminder, if you want it to be permanent, do it in www.burlesque-expo.com, as anything else will eventually get overwritten).  The way this works is:
- the "View" is effectively the use case you're trying to change - for these cases, it's all coded as MakeWhatever - as in MakeAct, MakeClass, Make Persona, Make Troupe, etc.  All the "make" stuff is using common code, so it all works the same.  
    -  Then the "Code" is what I showed with the bubbles on the image above - it's how my software knows where to put what text you give it.  
    - "Summary" is an attempt to let non-SW developers write a note to themselves that makes more sense - it's not visible anywhere but here.  
    - "Description" - is what actually shows up on the page.


2 of these are new (previously hard coded), and I'll need to sync up the update of this ticket to make it so.

For the other stuff (all the labels & help texts), it's a real PITA to pull them out of code and into editability... but they are very easy to change.  They all live here:

https://github.com/bethlakshmi/gbe-divio-djangocms-python2.7/blob/master/gbe_forms_text.py

And I'm glad to do the updates once we agree on terms that need to change.

**Comment by bethlakshmi** [comment]
**Fix the flow.**

For returning users, our flow isn't so bad - we have all the account stuff we need, their bios may not have changed, so really the act/class creation stuff is what needs to be done.  What really sucks is being a new user (which should actually be the 80% case if we want the conference to grow).  That flow is PAINFUL.  Right now it looks like:

| Step               | Action             | Image            | 
| ------------- | ------------- | ------------- |
| 1                     |  Click on a "submit your act" type link | Get Account Create Form: ![Screen Shot 2022-09-21 at 12 23 30 PM](https://user-images.githubusercontent.com/4542707/191559530-4bc0c11c-9bb7-4a1f-8533-6123a5b33503.png) | 
| 2                    | Submit account create form from step 1  by pressing "Register" | Get Profile Form:  ![Screen Shot 2022-09-22 at 8 25 38 AM](https://user-images.githubusercontent.com/4542707/191746351-3c37f8a8-ae35-4745-8085-ed65fc99cab2.png)![Screen Shot 2022-09-22 at 8 25 53 AM](https://user-images.githubusercontent.com/4542707/191746382-92cf3867-6bc6-42a6-bc90-3dfabf0ebd27.png) |
| 3                    | Submit from form step 2 by pressing "Update my Account" | Get Persona Form (or Troupe):  ![Screen Shot 2022-09-22 at 8 29 21 AM](https://user-images.githubusercontent.com/4542707/191747513-66dea42c-2003-4b71-a41e-e516ab11f07c.png) ![Screen Shot 2022-09-22 at 8 29 29 AM](https://user-images.githubusercontent.com/4542707/191747541-22f1765a-d54a-4d07-9e99-e698b55fe3e6.png) |
| 4                    | Submit from form step 3 by pressing "Save" | Get Act Submit Form (or Class Submit Form):  ![Screen Shot 2022-09-22 at 8 36 43 AM](https://user-images.githubusercontent.com/4542707/191748849-f0e71b4c-9598-48bd-99a1-9b3245ec5a64.png) ![Screen Shot 2022-09-22 at 8 37 08 AM](https://user-images.githubusercontent.com/4542707/191748913-e140b1a2-5941-419f-84bc-c5906ba3ddec.png) |
| 5                    | Choose "Proceed to Payment" (we actually loose folks on this, they save drafts and don't come back |  Get the payment form (no screen available). |
| 6                    | Do whatever paypal makes you do (that, at least, should be familiar) |

Luna - were you saying, let's simplify this?  If so, I agree.  It's written in the same way our underlying data structures look.  We have an account, a profile, a persona, an act, and then the bunch of paymenty stuff.  That's good, to a point, but 6 screens of a "wizard" with no sense of place strikes me as painful.

And half of these fields are absolutely not required, and they are sprinked in with everything else.

My proposal (so glad for this to be disagreed with, though...):
1 - combine account and profile (steps 1 & 2) - always, for everyone in every flow.  For new users, they get 1 form with both screens, for users updating their profile, they don't get the password update stuff (it's more involved and can be done, but deserves another ticket if we want that).  So no real difference on update.
2 - If the user doesn't have a persona, put creation on the same page as act/class creation.  Make there be 2 sections:
       - Performer/Teacher info (ie persona) (step 3) - with some nice intro text about what this is and how we use it (one thing I'd strongly recommend - mention that the Bio is what we publish online if they are accepted, and that the Stage Name is how our staff and MC will be addressing/introducing them, as well as the heading on their bio on our site.
       - Act/Class info (step 4)
3 - Could we remove phone number as a required profile field (it feels weirdly intrusive for all users to give us this, given how rarely we use it), and can we put it into Act/Class submission (ie, step 4 or whatever that becomes) - we REALLY want this for performers/teachers because they get flaky and we need to contact them.  But that would let us also highlight (w. help text) that if you are selected and we don't hear from you, we're gonna call and text you to see what's going on.  I can store it in the same place I already do, this would just fix flow.
4 - Should we do things to make the required vs. optional fields flow better - I see "required" data sprinkled throughout, and that feels like cognitive load to me.  Deferring totally to Luna here... this is not my speciality.  But I could pretty easily break optional stuff out, and do all kinds of visual tricks - stick in a colored card to separate it, put it in an expandable section (is it an accordion if it only has one fold?).  That makes it less likely we'll get that stuff, but it also makes the form less daunting.
5 - Wizardify it - when we know that someone is clicking on a "submit act" or "submit class" link, give them a sense of place.  I could do a "step 1 of 6" at the top.  I would happily accept recommendations for look and feel of that and how to make it the right level of prominence.  Also - if someone points me at a widget they like, I could a visual like this:  https://devrecipes.net/step-progres-bar-with-pure-css/    --- another point of wizardification is to change the bottom action buttons to classic wizardy words - Back/Next/Cancel - with the caveat that "cancel" just bumps out of of this screen (it doesn't erase data from previous flows... doing THAT would be big work I can't do right now)
6 - Just a personal thing - can we change "Experience" to "started performing in (year)"?  And does it have to be required?  Reason I want it to be year people started is - I can calculate experience from that dynamically, and then folks never need to update it.  I always forget, and I bet you do too, Scratch - so our years of experience in Burlesque are likely crazy out of date.  If we don't remember... who else will?

**Comment by bethlakshmi** [comment]
Note to self - also clean up error messages - they look pretty F'ed up with the new layout:
![Screen Shot 2022-09-22 at 9 07 02 AM](https://user-images.githubusercontent.com/4542707/191755245-6736f2c8-a17f-4ea6-8c94-142eb488560e.png)

**Comment by bethlakshmi** [comment]
OK - I fixed my ugly error messages.

AND - I think I fixed most of the layout stuff that I was bothered by.  Bootstrap has some nice [form layout stuff](https://getbootstrap.com/docs/4.0/components/forms/) (we already use Bootstrap)... which helped me:
- make the help text small
- keep the layout as originally designed
- remove a bunch of crap that no one sees which make the code hard for me to read! (yippeee!)

What do ya'll think?

**Comment by bethlakshmi** [comment]
OK, all the updates I intend to make on this ticket are done.  

FYI - 
- I use the label PR-ready as my short hand to remind myself to review the code at a later date (usually ~2-3 days later, this will be probably Friday night). It means the work is done, I'm just taking a breath to make sure I did a decent job.
- On Test Server - means it's the view that's installed on the test server right now.  This gets tricky, because I can only put on 1 branch at a time, and the best practice is to isolate them.  So this is a 1 ticket at a time type thing.  I may play fast and loose with best practices on this, and if it seems low risk enough, chain branches together so that I can have more than 1 ticket live on test for shared view.

Once the code is merged with our main line, I'll install in on production, and it will be on all servers all the time.

**Comment by bethlakshmi** [comment]
Finished the easy stuff.  Released on live
